dask_snowflake/tests/test_core.py

```diff
@@ -64,6 +64,21 @@ def test_write_read_roundtrip(table, connection_kwargs, client):
     )
 
 
+def test_read_empty_result(table, connection_kwargs, client):
+    # A query that yields in an empty results set should return an empty DataFrame
+    to_snowflake(ddf, name=table, connection_kwargs=connection_kwargs)
+
+    result = read_snowflake(
+        f"SELECT * FROM {table} where A > %(target)s",
+        execute_params={"target": df.A.max()},
+        connection_kwargs=connection_kwargs,
+        npartitions=2,
+    )
+    assert type(result) is dd.DataFrame
+    assert len(result.index) == 0
+    assert len(result.columns) == 0
```

Comment:
It'd be nice if we could match the columns in the empty dataframe, but I'll file that in the I-also-want-a-pony feature list.
![image](https://user-images.githubusercontent.com/1163939/200933810-714575e3-652e-443b-b2aa-bb1c131923aa.png)